auth/jwt_controller.go

```diff
@@ -21,18 +21,39 @@ import (
 // the "auth.jwt.secret" config entry.
 // The token is set to expire in the amount of seconds defined by
 // the "auth.jwt.expiry" config entry.
-func GenerateToken(id interface{}) (string, error) {
+func GenerateToken(username interface{}) (string, error) {
 	expiry := time.Duration(config.GetInt("auth.jwt.expiry")) * time.Second
 	now := time.Now()
 	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
-		"userid": id,
+		"userid": username,
 		"nbf":    now.Unix(),             // Not Before
 		"exp":    now.Add(expiry).Unix(), // Expiry
 	})
 
 	return token.SignedString([]byte(config.GetString("auth.jwt.secret")))
 }
 
+// GenerateTokenWithClaims generates a new JWT with custom claims.
+// The token is created using the HMAC SHA256 method and signed using
+// the "auth.jwt.secret" config entry.
+// The token is set to expire in the amount of seconds defined by
+// the "auth.jwt.expiry" config entry.
+func GenerateTokenWithClaims(username interface{}, claims jwt.MapClaims) (string, error) {
+	expiry := time.Duration(config.GetInt("auth.jwt.expiry")) * time.Second
+	now := time.Now()
+	customClaims := claims
+	customClaims["userid"] = username
+	customClaims["nbf"] = now.Unix()             // Not Before
+	customClaims["exp"] = now.Add(expiry).Unix() // Expiry
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, customClaims)
+
+	return token.SignedString([]byte(config.GetString("auth.jwt.secret")))
+}
+
+// TokenFunc is the function used by JWTController to generate tokens
+// during login process.
+type TokenFunc func(username interface{}, user interface{}) (string, error)
```

Comment:
First parameter should be `*goyave.Request`.
```suggestion
type TokenFunc func(request *goyave.Request, user interface{}) (string, error)
```